### PR TITLE
refactor: decouple preserved and in-mem catalog (part 2)

### PR DESCRIPTION
**Pull request by crepererum**
For #1740.

# Outlook
- move type parameters around / API clean-up (this PR)
- move in-mem catalog out of the transaction, this will be a HUGE simplification (next PR)
- cleanups (e.g. remove some `Arc`s that are no longer needed) (the PR after)

### CODE REVIEW

File: parquet_file/src/catalog.rs [hunk 1]
```diff
@@ -352,18 +296,69 @@ where
...
     /// Create new catalog w/o any data.
     ///
     /// An empty transaction will be used to mark the catalog start so that concurrent open but still-empty catalogs can
     /// easily be detected.
-    pub async fn new_empty(
+    pub async fn new_empty<S>(
         object_store: Arc<ObjectStore>,
         server_id: ServerId,
-        db_name: impl Into<String> + Send,
+        db_name: String,
```
Comment (tustvold) [inline parquet_file/src/catalog.rs:356]: What is the reason for this change?

**Review by tustvold — APPROVED** [review]
Looks good to me - I like where this is going :+1:

File: parquet_file/src/catalog.rs (hunk 1 above)
Comment (crepererum) [reply parquet_file/src/catalog.rs:356]: Mostly simplifications (we eventually convert it to a string anyway) and because mixing `impl...` and type params don't mix well. Aka:

1. You can not use `new_empty::<T>(...)` if one of the params is an `impl`, because Rust then doesn't seem to know which type you mean.
2. So you have to pull every `impl` into a real type param (aka `new_empty<S, N>(...) where S: ..., N: Into<String> + Send`).
3. But then every call to that function gets more complicated, because now you have to write `new_empty::<S, _>(...)`

Comment (tustvold) [reply parquet_file/src/catalog.rs:356]: Oh yeah, forgot about that - makes sense :+1:

**Review by alamb — COMMENTED** [review]
I skimmed these changes and they look like a nice improvement to me
